prelude/toolchains/cxx.bzl

```diff
@@ -170,49 +163,29 @@ def _system_cxx_toolchain_impl(ctx: AnalysisContext):
         CxxPlatformInfo(name = "x86_64"),
     ]
 
-def _windows_linker_wrapper(ctx: AnalysisContext, linker: cmd_args) -> cmd_args:
-    # Linkers pretty much all support @file.txt argument syntax to insert
-    # arguments from the given text file, usually formatted one argument per
-    # line.
-    #
-    # - GNU ld: https://gcc.gnu.org/onlinedocs/gcc/Overall-Options.html
-    # - lld is command line compatible with GNU ld
-    # - MSVC link.exe: https://learn.microsoft.com/en-us/cpp/build/reference/linking?view=msvc-170#link-command-files
-    #
-    # However, there is inconsistency in whether they support nesting of @file
-    # arguments inside of another @file.
-    #
-    # We wrap the linker to flatten @file arguments down to 1 level of nesting.
-    return cmd_script(
-        ctx = ctx,
-        name = "windows_linker",
-        cmd = cmd_args(
-            ctx.attrs.linker_wrapper[RunInfo],
-            linker,
-        ),
-        os = ScriptOs("windows"),
-    )
+def _run_info(args):
+    return None if args == None else RunInfo(args = [args])
+
+def _get_default_compiler() -> str:
+    os = host_info().os
+    if os.is_windows:
+        return "prelude//toolchains/msvc:msvc_tools"
+    else:
+        return "prelude//toolchains/cxx/clang:path_clang_tools"
 
 system_cxx_toolchain = rule(
     impl = _system_cxx_toolchain_impl,
     attrs = {
+        "compiler": attrs.exec_dep(providers = [NativeCompiler], default = _get_default_compiler()),
         "c_flags": attrs.list(attrs.string(), default = []),
-        "compiler": attrs.string(default = "cl.exe" if host_info().os.is_windows else "clang"),
```

Comment:
This is a breaking change for anyone who uses these attributes in `system_cxx_toolchain`. I don't think we should remove them.

Reply:
I reintroduced the attributes and I think the end result is fully backwards compatible. I did some tests on Linux and Windows using those parameters.